src/inference/src/dev/icompiled_model.cpp

```diff
@@ -50,6 +50,21 @@ ov::ICompiledModel::ICompiledModel(const std::shared_ptr<const ov::Model>& model
 
         std::unordered_map<std::shared_ptr<ov::descriptor::Tensor>, std::shared_ptr<ov::descriptor::Tensor>> tensor_map;
         for (const auto& param : model->get_parameters()) {
+            {
+                const auto& p_shape = param->get_partial_shape();
+                const auto& layout = param->get_layout();
+                if (p_shape.rank().is_static() && p_shape.rank() != 0 && ov::layout::has_batch(layout)) {
```

Comment:
IMO, invalid shapes should be checked either on model validation level or on plugin level.